packages/expo-modules-core/ios/AppDelegates/ExpoAppInstance.swift

```diff
@@ -0,0 +1,86 @@
+import React_RCTAppDelegate
+
+private var reactDelegateHandlers = [ExpoReactDelegateHandler]()
+
+@objc(EXAppInstance)
+open class ExpoAppInstance: RCTAppDelegate {
+  @objc
+  public let reactDelegate = ExpoReactDelegate(handlers: reactDelegateHandlers)
+
+  @objc
+  open override func createRootViewController() -> UIViewController {
+    return reactDelegate.createRootViewController()
+  }
+
+  @objc
+  open override func createRCTRootViewFactory() -> RCTRootViewFactory {
+    let bundleUrlBlock: RCTBundleURLBlock = { [weak self] in
+      return self?.bundleURL()
+    }
+
+    let configuration = RCTRootViewFactoryConfiguration(
+      bundleURLBlock: bundleUrlBlock,
+      newArchEnabled: newArchEnabled(),
+      turboModuleEnabled: turboModuleEnabled(),
+      bridgelessEnabled: bridgelessEnabled()
+    )
+
+    configuration.createRootViewWithBridge = { bridge, moduleName, initProps in
+      return self.createRootView(with: bridge, moduleName: moduleName, initProps: initProps)
+    }
+
+    configuration.createBridgeWithDelegate = { delegate, launchOptions in
+      return self.createBridge(with: delegate, launchOptions: launchOptions)
+    }
+
+    configuration.customizeRootView = { rootView in
+      // @tsapeta: We cannot just call `self.customize(rootView)` – see the description of the below method.
```

Comment:
the description is not at below?

Reply:
It is a doc comment in its header, see https://github.com/expo/expo/pull/33348/files#diff-5fe3899fbb747f7f3bc720480cb516169aea508e399dbb8a2de07f999f11104dR20-R25
Should I put it in two places? I think we should upstream this change anyway – RCTRootView shouldn't be there.